dali/python/nvidia/dali/_conditionals.py

```diff
@@ -43,584 +43,1159 @@
 
 from enum import Enum
 
-import tree
-
+def check_nesting_support():
+    """Function to check if nesting conditionals is supported in current environment.
+    
+    Note:
+        This is due to lack of support for dm-tree package in Python 3.6. When Python 3.6
+        support is dropped this can be removed.
+
+    Returns:
+        _bool: Is nesting conditionals supported.
+    """
+    try:
+        import tree
+        return True
+    except Exception as e:
+        return False
+    
+    
+if check_nesting_support():
```

Comment:
As I understand this turns off the conditional execution for python3.6 now, right?
I'm afraid we cannot do that.

Reply:
It does not. As I stated in the comments for Python 3.6 I pasted old implementation of this.

Reply:
https://github.com/NVIDIA/DALI/pull/4888#discussion_r1213984747

Reply:
Oh. There is else branch. Hard to spot it. Maybe there should be an another module, called legacy and we should import either one or the second to avoid messing up with code?
```
_conditionals.py:

if check_nesting_support():
    from _conditionals_new import *
else:
    from _conditionals_legacy import *
    # print waring about lack of support for nested data strucutres
```

Reply:
Done